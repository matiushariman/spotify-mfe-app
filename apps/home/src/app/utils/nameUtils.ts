export const displayTrackName = (trackName: string) => {
  const MAX_TRACK_NAME_LENGTH = 15;

  if (trackName.length <= MAX_TRACK_NAME_LENGTH) {
    return trackName;
  }

  return `${trackName.slice(0, MAX_TRACK_NAME_LENGTH)}...`;
};

export const displayArtistName = (artistName: string) => {
  const MAX_ARTIST_NAME_LENGTH = 23;

  if (artistName.length <= MAX_ARTIST_NAME_LENGTH) {
    return artistName;
  }

  return `${artistName.slice(0, MAX_ARTIST_NAME_LENGTH)}...`;
};
