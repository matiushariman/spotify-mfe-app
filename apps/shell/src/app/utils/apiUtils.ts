import type { ApiErrorResponse } from '@spotify-mfe-app/api-client';
import { removeAuthToken } from '@spotify-mfe-app/auth-utils';

export const handleApiError = (err: ApiErrorResponse) => {
  const {
    error: { status },
  } = err;

  if (status === 401) {
    window.alert(
      'Your session has expired! Click close to reauthenticate your session.'
    );

    removeAuthToken();
    window.location.replace('/');
  }
};
