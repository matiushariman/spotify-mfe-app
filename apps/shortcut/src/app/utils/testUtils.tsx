import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { QueryClientProvider, QueryClient } from 'react-query';

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => (
  <QueryClientProvider client={new QueryClient()}>
    {children}
  </QueryClientProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { customRender as render, AllTheProviders as wrapper };
