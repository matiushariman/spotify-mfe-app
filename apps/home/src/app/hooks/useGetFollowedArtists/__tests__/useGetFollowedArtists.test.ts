import { useGetFollowedArtists } from '..';

import { renderHook, wrapper, waitFor } from '../../../utils/testUtils';

describe(useGetFollowedArtists, () => {
  it('should return followed artists response', async () => {
    const { result } = renderHook(() => useGetFollowedArtists(), {
      wrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
