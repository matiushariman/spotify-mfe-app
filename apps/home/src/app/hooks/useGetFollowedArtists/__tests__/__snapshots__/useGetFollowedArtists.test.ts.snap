// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useGetFollowedArtists should return followed artists response 1`] = `
Object {
  "artists": Object {
    "cursors": Object {
      "after": null,
    },
    "href": "https://api.spotify.com/v1/me/following?type=artist&limit=20",
    "items": Array [
      Object {
        "external_urls": Object {
          "spotify": "https://open.spotify.com/artist/6k0IBR0lU42s2GYpNX7kA9",
        },
        "followers": Object {
          "href": null,
          "total": 1342132,
        },
        "genres": Array [
          "contemporary r&b",
          "dance pop",
          "neo soul",
          "quiet storm",
          "r&b",
          "urban contemporary",
        ],
        "href": "https://api.spotify.com/v1/artists/6k0IBR0lU42s2GYpNX7kA9",
        "id": "6k0IBR0lU42s2GYpNX7kA9",
        "images": Array [
          Object {
            "height": 640,
            "url": "https://i.scdn.co/image/ab6761610000e5ebccd9085c5dbc5f5a991ae59c",
            "width": 640,
          },
          Object {
            "height": 320,
            "url": "https://i.scdn.co/image/ab67616100005174ccd9085c5dbc5f5a991ae59c",
            "width": 320,
          },
          Object {
            "height": 160,
            "url": "https://i.scdn.co/image/ab6761610000f178ccd9085c5dbc5f5a991ae59c",
            "width": 160,
          },
        ],
        "name": "Brian McKnight",
        "popularity": 63,
        "type": "artist",
        "uri": "spotify:artist:6k0IBR0lU42s2GYpNX7kA9",
      },
    ],
    "limit": 20,
    "next": null,
    "total": 0,
  },
}
`;
